src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl

```diff
@@ -583,10 +528,9 @@ function update_aux_diffusive!(dg::DGModel, m::HBModel, Q::MPIStateArray, t::Rea
   # update vertical diffusivity for convective adjustment
   function f!(m::HBModel, Q, A, D, t)
     @inbounds begin
-      A.w = -(D.∇u[1,1] + D.∇u[2,2])
+      # probably need to be smarter about factoring out viscosity
+      A.w = -(D.ν∇u[1,1] + D.ν∇u[2,2]) / m.νʰ
```

Comment:
@glwagner so right now I can get away with this because we have a diagonal viscosity tensor. Is there a reliable way to do this for an arbitrary viscosity tensor? Combining viscosity and `∇u` into a single variable is very convenient for the boundary condition specification.

Reply:
Not sure I completely understand --- what is `A.w`? Is the dilemma between using the velocity gradient, versus the diffusive flux components, as prognostic variables in the first-order formulation?

PS: spaces after commas!

Reply:
Ah, so `w = ∫(-∇⋅u)dz`. This is storing in `-(∂u/∂x + ∂v/∂y)` to be used as the integrand for the calculation. I just switched from storing just `∇u` to storing `ν∇u` and I'm trying to recover `∂u/∂x` and `∂v/∂y`. This is currently easy because `ν = Diagonal([νʰ,νʰ,νᶻ])`, so I can just divide out `νʰ` from `ν∇u[1,1]` and `ν∇u[2,2]` to get `∂u/∂x` and `∂v/∂y`. 

I'm wondering if there's an easy way to future proof this or are we just doing to need to do something like `∇u = ν \ ν∇u` with a temporary variable and then use the components of `∇u`. (I might have just answered my own question with that)

Also for the purposes of this, you can ignore everything before the `.` in a variable as that's just specifying where in memory it's stored.

Reply:
Got it! As you say I do think that in general you will wind up having to invert a matrix to obtain the velocity components from the components of the viscous momentum flux.

In general, the viscosities will not only be tensorial, but will be nonlinear functions of the velocity and density field.

Also, velocity gradients may be used elsewhere --- mostly in the calculation of the nonlinear viscosities and diffusivities themselves (in parameterizations), and probably also for post-processing.

The general need to access velocity gradients in places other than this line might warrant formulating the model in terms of velocity gradients themselves rather than viscous momentum flux, depending, of course, on how complicated that makes implementing boundary conditions.

Reply:
As we discussed on slack, we'll assume we can properly pass the velocity gradients into the parameterizations and continue using the momentum fluxes here.